### ISSUE TITLE
Interface LDR with BB-WI

### ISSUE THREAD

**Issue by zrajani** [zrajani/iLETTERBOX_CAPSTONE_PROJECT_EMBT_F2020#21]
This is the last phase of hardware interfacing in our project. In this interfacing, we want to interface the LDR module with BB-WI. With the help of this sensor, we shall make the letter count as zero so that it marks that we have removed all the letters from the letterbox.

**Comment by zrajani** [comment]
All issues faced during this interface shall be mentioned here. Please post the issues along with the ways one can use to resolve them.

**Comment by zrajani** [comment]
This interfacing was considered to be quite simpler than the rest of the interfacing as it was just adding another sensor via the GPIO Digital Communication protocol. The only one issue that was faced was by adjusting the sensitivity level of the sensor. For the project since we had to reset the counter when either darkness or brightness was there (our choice which condition should reset the count of the letters). The potentiometer was turned by human action or don't know how it never used to go low and never detected darkness. 

Thus, to resolve this issue up we had to make the change in the potentiometer that was attached to the module. This was done in steps and employing trial and error method till we were satisfied that the sensor is detecting the darkness or brightness in a way which it should do.

**Comment by zrajani** [comment]
An issue was being faced as to when the sensor detected brightness or darkness as this was crucial in resetting the count of letters. Thus, with us what had happened was that if we set the condition in the code stating that if the IR sensor is low (object present) and brightness high (LDR pin high) then keeping on counting and the opposite condition would reset the counter. Now if we look at it properly we were right logically but we didn't keep in mind that the LDR is an active low sensor like the IR sensor thus we resolved it and got a proper count of 0 by giving the same condition but the program mentioning that if both the pins are at low voltage level then count the letters i.e. increment the count else make the counter 0

**Comment by zrajani** [comment]
Since we always try to integrate all the sensors working together thus we faced a small bug in the integration of all the sensors. When we combined the setup for this issue along with Issue #19 ,  #12 we faced some issues that are suddenly the if loops started to run twice like if the letter was sensed it used to count an increment of 1 and again increment the count when the GSM code for sending SMS was completed. This was not good as it may give the wrong count for our limitation that only a letter is inserted in the box.

**Comment by zrajani** [comment]
> Since we always try to integrate all the sensors working together thus we faced a small bug in the integration of all the sensors. When we combined the setup for this issue along with Issue #19, #12 we faced some issues that are suddenly the if loops started to run twice like if the letter was sensed it used to count an increment of 1 and again increment the count when the GSM code for sending SMS was completed. This was not good as it may give the wrong count for our limitation that only a letter is inserted in the box.

To get the issue resolved we did two things:
1.  We read the program carefully and understood that we didn't declare the usage of the pin P9.41 usage which we thought might be the cause of error thus we rectified the same 
2. We removed all the connections and re-connected the circuit back thinking some wiring might be loose or improper

Thus both the steps are taken and when the code was under re-run the code compiled successfully and ran as per expectation. Thus we assume that it was a wiring mistake and coding error as well.

**Comment by zrajani** [comment]
During the meeting, we were asked to get the SMS alerted to the user when letters were removed and also print on the console the count of letters. During this change, we were able to send the SMS when all letters were removed but faced small difficulty in getting the count printed on the console. It was found since the GSM code required time to run thus it was not getting printed in time. Thus when the line of printing the count was just shifted before the GSM code can execute our problem was resolved and thus we can now print the number of letters deposited on the console and also print the count after we remove all the letters that we print the count =0 when all the letters are removed from the letterbox. The issue was resolved on the same day during the meeting
